cmip6_downscaling/data/cmip.py

```diff
@@ -272,11 +285,97 @@ def gcm_munge(ds: xr.Dataset) -> xr.Dataset:
     return ds
 
 
-def get_gcm_grid_spec(gcm: str) -> str:
-    gcm_grid = load_cmip(
+def get_gcm(
+    gcm: str,
+    scenario: str,
+    variables: Union[str, List[str]],
+    train_period_start: str,
+    train_period_end: str,
+    predict_period_start: str,
+    predict_period_end: str,
+    chunking_approach: Optional[str] = None,
+    cache_within_rechunk: Optional[bool] = True,
+) -> xr.Dataset:
+    """
+    Load and combine historical and future GCM into one dataset.
+
+    Parameters
+    ----------
+    gcm: str
+        Name of GCM
+    scenario: str
+        Name of scenario
+    variables: str or list
+        Name of variable(s) to load
+    train_period_start: str
+        Start year of train/historical period
+    train_period_end: str
+        End year of train/historical period
+    predict_period_start: str
+        Start year of predict/future period
+    predict_period_end: str
+        End year of predict/future period
+    chunking_approach: Optional[str]
+        'full_space', 'full_time', or None
+
+    Returns
+    -------
+    ds_gcm: xr.Dataset
+        A dataset containing both historical and future period of GCM data
+    """
+    historical_gcm = load_cmip(
+        activity_ids='CMIP',
+        experiment_ids='historical',
+        source_ids=gcm,
+        variable_ids=variables,
+        return_type='xr',
+    ).sel(time=slice(train_period_start, train_period_end))
+    future_gcm = load_cmip(
+        activity_ids='ScenarioMIP',
+        experiment_ids=scenario,
         source_ids=gcm,
+        variable_ids=variables,
         return_type='xr',
-    ).isel(time=0)
+    ).sel(time=slice(predict_period_start, predict_period_end))
+    ds_gcm = xr.combine_by_coords([historical_gcm, future_gcm], combine_attrs='drop_conflicts')
+
+    if chunking_approach is None:
+        return ds_gcm
+
+    if cache_within_rechunk:
+        path_dict = {
+            'gcm': gcm,
+            'scenario': scenario,
+            'train_period_start': train_period_start,
+            'train_period_end': train_period_end,
+            'predict_period_start': predict_period_start,
+            'predict_period_end': predict_period_end,
+            'variables': variables,
+        }
+        rechunked_path = make_rechunked_gcm_path(chunking_approach=chunking_approach, **path_dict)
```

Comment:
question. Do you think we should bundle the rechunking step in this function, or rechunk after it's returned?

Reply:
hmm either way works i think -- bundling reduces the number of tasks in the actual flow itself, which i thought is nice because every function in the flow has to be a task. So if we were to split it out, the rechunking and make_path functions both have to be wrapped into a task. It's not difficult but feels less clean to me. Any rationale to split it out?

Reply:
Sounds good!  Less repeated tasks definitely seems clearer.